dsgrid/config/date_time_dimension_config.py

```diff
@@ -35,12 +41,44 @@ def check_dataset_time_consistency(self, load_data_df, time_columns):
         time_range = time_ranges[0]
         # TODO: need to support validation of multiple time ranges: DSGRID-173
 
+        assert (
+            load_data_df.schema[time_col].dataType == TimestampType()
+        ), f"datetime {time_col} column must be TimestampType"
+
         expected_timestamps = time_range.list_time_range()
+        if self.model.datetime_format.format_type in [DatetimeFormat.LOCAL_AS_STRINGS]:
+            if "id" in load_data_df.columns:
+                groupby = ["id"]
+            else:
+                groupby = [x for x in load_data_df.columns if x not in [time_col, VALUE_COLUMN]]
+            start_times = load_data_df.groupBy(*groupby).agg(F.min(time_col).alias(time_col))
+            min_ts = start_times.select(F.min(time_col).alias(time_col)).collect()[0][time_col]
+            start_times = start_times.withColumn("min_ts", F.lit(min_ts))
+            time_deltas = start_times.select(
+                *groupby, (F.col(time_col) - F.col("min_ts")).alias("time_delta")
+            )
+            timestamps = load_data_df.select(*groupby, time_col).join(time_deltas, groupby, "left")
+            timestamps = timestamps.select(
+                *groupby, (F.col(time_col) - F.col("time_delta")).alias("aligned_time")
+            )
+            timestamp_counts = [
+                x["count"] for x in timestamps.groupBy("aligned_time").count().collect()
+            ]
+            if len(set(timestamp_counts)) != 1:
+                raise DSGInvalidDataset(
+                    f"load_data {time_col}s do not have the same time range when adjusting for offsets."
+                )
+            if len(timestamp_counts) != len(expected_timestamps):
+                raise DSGInvalidDataset(
+                    f"load_data {time_col}s do not have the expected number of timestamps when adjusting for offsets."
+                )
+            return
+
         actual_timestamps = [
             x[time_col].astimezone().astimezone(tz)
-            for x in load_data_df.select(time_col)
+            for x in load_data_df.filter(f"{time_col} is not null")
```

Comment:
I think you suggested moving up the filter last time.

Reply:
Somewhere along the way I wrote something like "select -> filter -> distinct." Spark can probably optimize this, but I read somewhere that you should be clear in case it can't. It's definitely better to select one column before filter.